### PR TITLE
arbitrary numbers of features added, random feature init assignment t…

### PR DESCRIPTION
**Pull request by Riley16**
…ested, random weight assignment tested, optional finite horizon parameter added to wrapper.eval_episodes, finite horizon added to SCOT candiate trajectories to ensure termination of SCOT algorithm

### CODE REVIEW

File: env.py [hunk 1]
```diff
@@ -41,24 +41,28 @@ def __init__(self, height:int, width:int, gamma:float, white_r:float=None,
         self.nS = width*height
         self.actions_to_grid = {a: g for a, g in enumerate(ACTIONS)}
         self.grid_to_actions = {g: a for a, g in enumerate(ACTIONS)}
+        if noise is not None:
```
Comment (vliu15) [inline env.py:45]: unnecessary if else

File: env.py [hunk 2]
```diff
@@ -11,7 +11,7 @@ class Grid(object):
     '''
 
     def __init__(self, height:int, width:int, gamma:float, white_r:float=None,
-                features_sq:List[Dict]=None, gen_features:List[List]=None, n_features:int=None,
+                features_sq:List[Dict]=None, gen_features:Union[List[List], Any]=None, n_features:int=None,
```
Comment (vliu15) [inline env.py:14]: change 'Any' to 'str'

**Review by Sradzi13 — APPROVED** [review]
Looks great! Just looked over the minor changes and tested them out last night :)

File: env.py (hunk 2 above)
Comment (Sradzi13) [reply env.py:14]: I think it is fine to leave it generalized but feel free to change it Vincent. Just trying to push for functionality and cleanliness rn

**Pull request merged**